clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -467,6 +468,7 @@ boolean joinGroupIfNeeded(final Timer timer) {
                 final RuntimeException exception = future.exception();
 
                 resetJoinGroupFuture();
+                rejoinReason = exception.getMessage();
```

Comment:
I wonder if we should put a bit more information here. For instance, we could say `rebalance failed due to %s error: %s` where the first `%s` would be the exception's class and the second the message. What do you think?